pkgs/applications/editors/xournalpp/default.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+
+, cmake
+, gettext
+, wrapGAppsHook
+, pkgconfig
+
+, glib
+, gsettings-desktop-schemas
+, gtk3
+, hicolor-icon-theme
+, libsndfile
+, libxml2
+, pcre
+, poppler
+, portaudio
+, zlib
+
+# Plugins don't appear to be working in this version, so disable them by not
+# building with Lua support by default. In a future version, try switching this
+# to 'true' and seeing if the top-level Plugin menu appears.
+, withLua ? false, lua
+}:
+
+stdenv.mkDerivation rec {
+  name = "xournalpp-${version}";
+  version = "1.0.8";
+
+  src = fetchFromGitHub {
+    owner = "xournalpp";
+    repo = "xournalpp";
+    rev = version;
+    sha256 = "01q84xjp9z1krna10gjj562km6i3wdq8cg7paxax1k6bh52ryvf6";
+  };
+
+  # The top-level CMakeLists.txt has a file path that contains `/usr/bin/`,
+  # which `fixCmakeFiles` will helpfully replace with /var/empty. We disable
+  # that step, run it manually, and then un-patch that line prior to
+  # configuring.
+  dontFixCmake = true;
+  preConfigure = ''
+    fixCmakeFiles .
+    sed -e 's|var/empty/local/bin/xopp-recording.sh|usr/local/bin/xopp-recording.sh|g' CMakeLists.txt > CMakeLists.txt.tmp
```

Comment:
What is `xopp-recording.sh` in first place?

https://github.com/xournalpp/xournalpp/pull/446/files looks like it's some shell script shipped with xournalpp (which currently doesn't work on nixos either, as it hardcodes bash and assumes to be able to call some binaries).

Personally, I think this belongs to `/usr/libexec/xournalpp` instead of `/usr/local/bin` - maybe we can improve upstream in that regard? (and make the recording feature conditional in nixpkgs?)

Reply:
reading up on https://github.com/xournalpp/xournalpp/commit/d417619c26777fca6f80c7421cfc5a9d02b7e458, and after a fulltextsearch on latest master, it seems the scary xopp-recording.sh script isn't called anymore, but portaudio is used to record.

I opened https://github.com/xournalpp/xournalpp/pull/956 to remove it from the repo and `CMakeLists.txt`, so we shouldn't need to hack `CmakeLists.txt` anymore.

Can you add this patch to `patches`?

Reply:
Done!